monasca_rsyslog/client.py

```diff
@@ -46,13 +48,22 @@ def __init__(self):
             user_agent='rsyslog-monasca'
         )
         self._url = cfg.CONF.api.url
+        self.buffer = {}
+        self.interval = interval
+        self.start_time = time.time()
 
     def post_logs(self, data):
         """Post logs to Monasca which are suitably pre-encoded as JSON."""
-        
-        self._sess.post(
-            '/logs',
-            endpoint_override=self._url,
-            headers={ 'Content-Type': 'application/json' },
-            data=data
-        )
+        for key, value in jsonutils.loads(data).items():
+            self.buffer.setdefault(key, []).extend(value)
+        elapsed_time = time.time() - self.start_time
+        if elapsed_time > self.interval:
```

Comment:
I think this works fine if you have a steady inflow of logs, but what if you have a single critical log file that comes in, and then nothing for a few hours. Won't that critical log file sit in the buffer until another log message comes in? Do we need to periodically flush this, so that we can guarantee that logs won't sit in the buffer for much longer than the interval?

Reply:
Any thoughts on this?

Reply:
To answer your question, I have put in a timeout mechanism using a non configurable `proc_interval` of 1 second in the `stdin_by_lines` function which yields None and allows items in the queue to be flushed if the acceptable interval has elapsed before it is called again.